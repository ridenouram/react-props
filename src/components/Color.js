import React from 'react';
import PropTypes from 'prop-types';

function Color({ name, hex, rgb }) {

  const divStyle = {
    backgroundColor: hex,
  };

  return (
    <dl style={divStyle}>
      <dt>Name</dt>
      <dd>{name}</dd>

      <dt>Hex</dt>
      <dd>{hex}</dd>

      <dt>RGB</dt>
      <dd>red: {rgb.red}</dd>
      <dd>green: {rgb.green}</dd>
      <dd>blue: {rgb.blue}</dd>
    </dl>
  );
}

Color.propTypes = {
  name: PropTypes.string.isRequired,
  hex: PropTypes.string.isRequired,
  rgb: PropTypes.object.isRequired
};

export default Color;

