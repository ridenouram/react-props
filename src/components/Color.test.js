import React from 'react';
import { shallow } from 'enzyme';
import Color from './Color';

describe('Header Component', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<Color name="yellow peach" hex="#deb993" rgb={{ red: 222, green: 185, blue: 147 }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
