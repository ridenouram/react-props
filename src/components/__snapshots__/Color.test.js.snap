// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component matches a snapshot 1`] = `
<dl
  style={
    Object {
      "backgroundColor": "#deb993",
    }
  }
>
  <dt>
    Name
  </dt>
  <dd>
    yellow peach
  </dd>
  <dt>
    Hex
  </dt>
  <dd>
    #deb993
  </dd>
  <dt>
    RGB
  </dt>
  <dd>
    red: 
    222
  </dd>
  <dd>
    green: 
    185
  </dd>
  <dd>
    blue: 
    147
  </dd>
</dl>
`;
